### PR TITLE
Support retrieving the status bar of Windows File Explorer (#6769)

### LINKED ISSUE #6769: Can't read status bar in File Explorer (Windows 10)
In File Explorer (WINDOWS+E), pressing NVDA+end (laptop: NVDA+shift+end) reads "No status line found", even if the status line is enabled (it's possible to disable it in the folder options.

Expected behaviour, read the status bar, for instance, "30 items, 1 selected, details view button, large icons view button".

### PR DESCRIPTION
**Pull request by JulienCochuyt**
### Link to issue number:

Fixes #6769

### Summary of the issue:

Currently, the Window File Explorer cannot be retrieved using the `reportStatusLine` global command.

### Description of how this pull request fixes the issue:

 - Take advantage of #4640 to implement custom status bar retrieval by the `explorer` app module.
 - Use a UIAWalker to retrieve it, as hinted by @josephsl in https://github.com/nvaccess/nvda/issues/6769#issuecomment-624532165
 - Still, instead of overlaying the status bar of its children for concerns expressed in https://github.com/nvaccess/nvda/issues/6769#issuecomment-833432110, allow app modules to provide their custom implementation of `api.getStatusBarText`
 - Implement `explorer.AppModule.getStatusBarText` as inspired by @ruifontes prototype in https://github.com/nvaccess/nvda/issues/6842#issuecomment-410670910 (hinted by @feerrenrut in https://github.com/nvaccess/nvda/issues/6769#issuecomment-411251846) but in a slightly more complex way - trying to cover more unexpected cases.

### Testing strategy:

 - Ran from source under Windows 10 20H2 19042.928 & 19042.964 both with a French locale.
 - Ensured expected result whether files are actually selected or not.

### Known issues with pull request:

This is my first UIA work ever: Review with caution!
I've tried to set safeguards on expectations, but there might be huge differences between Windows versions and I'm not able to test all of them.

### Change log entries:
New features
Changes
In Windows 10, the File Explorer status bar can now be retrieved using the standard gesture `NVDA+end` (desktop) / `NVDA+shift+end` (laptop).

For Developers
- Status bar text retrieval may now be customized by an AppModule.

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [ ] Unit tests.
- [ ] System (end to end) tests.
- [x] Manual tests.
- [ ] User Documentation.
- [x] Change log entry.
- [ ] Context sensitive help for GUI changes.

### CODE REVIEW

File: source/appModules/explorer.py [hunk 1]
```diff
@@ -354,6 +354,76 @@ def chooseNVDAObjectOverlayClasses(self, obj, clsList):
...
+			# We could not find the expected object. Resort to standard behavior.
+			raise NotImplementedError()
+		element = element.buildUpdatedCache(UIAHandler.handler.baseCacheRequest)
+		statusBar = UIA(UIAElement=element)
+		return statusBar
+
+	def getStatusBarText(self, obj) -> str:
+		if not isinstance(obj, UIA) or obj.UIAElement.cachedClassname != "StatusBarModuleInner":
+			# This is not the file explorer status bar. Resort to standard behavior.
+			raise not NotImplementedError()
```
Comment (michaelDCurran) [inline source/appModules/explorer.py:381]:
```suggestion
			raise NotImplementedError
```

**Review by michaelDCurran — CHANGES_REQUESTED** [review]
The UIA walker part looks fine to me.
The getStatusbarText code is a bit complex and I also have no idea how robust that would be on different Windows versions. Perhaps @josephsl  has some idea?
If this can be tested on Win 7, Win 8.1 and Win 10, then I'm good with this.

**Comment by lukaszgo1** [comment]
I've tested code from this PR on Windows 7, Server 2012, Windows 10 1511, 1809 and 20H1 and aside from Windows 7 which requires a pretty different implementation it works pretty well. In the `pr12371OldWindowsVers` branch of my fork based on this code I've implemented additional retrieval method for Windows 7, fixed a unnecessary garbage being reported when reading status bar in the Quick Access on Windows 10 and removed reporting about no files being selected. I personally consider this rather unnecessary since as far as I can tell there is no visual indication of this in the status bar, and files are unselected only when the folder is freshly opened or user unselects the current file himself Happy to hear a different opinions about this though as this is rather subjective. @michaelDCurran  Is the testing outlined above sufficient for you to continue with this?

**Comment by lukaszgo1** [comment]
@JulienCochuyt  Do you intent to continue this work?

**Comment by lukaszgo1** [comment]
@JulienCochuyt  Since this seemed more or less abandoned, and required changes to work on older Windows versions (which, as you've indicated in the PR description you're unable to test yourself) I've created #12845 which superseets this. I hope you do not mint.

**Comment by feerrenrut** [comment]
As per @lukaszgo1, I'm going to close this as abandoned and refer people to #12845

**Pull request closed without merging**